server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -269,13 +222,27 @@ public SourceFieldMapper build() {
                 }
             }
 
-            SourceFieldMapper sourceFieldMapper = new SourceFieldMapper(
-                sourceMode,
-                enabled.get(),
-                includes.getValue().toArray(Strings.EMPTY_ARRAY),
-                excludes.getValue().toArray(Strings.EMPTY_ARRAY),
-                indexMode
-            );
+            if ((mode.get() == Mode.SYNTHETIC || indexMode == IndexMode.TIME_SERIES)
```

Comment:
Do we need to check also for `IndexMode.LOGSDB` here?

I see we have tests in `resources/rest-api-spec/test/tsdb/20_mapping.yml` for `mode: time_series` but we are probably missing similar tests for LogsDB.

Reply:
Also instead of `mode.get()` shouldn't we use `sourceMode` here?

Reply:
As @martijnvg says we should not consider index mode here since it does not tell us the source mode anymore. I suspect this is related to the introduction of synthetic source in TSDB in 8.7 that we handle in other place too.

Reply:
Actually it looks like in that case we simply ignore includes and excludes https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java#L323.